src/integrationTest/java/org/opensearch/security/api/InternalUsersRegExpPasswordRulesRestApiIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+
+    static {
+        clusterSettings.put(ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_ERROR_MESSAGE, "xxx")
+            .put(ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, "(?=.*[A-Z])(?=.*[^a-zA-Z\\\\d])(?=.*[0-9])(?=.*[a-z]).{8,}")
+            .put(ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_STRENGTH, PasswordValidator.ScoreStrength.FAIR.name());
+    }
+
+    String internalUsers(String... path) {
+        final var fullPath = new StringJoiner("/").add(super.apiPath("internalusers"));
+        if (path != null) {
+            for (final var p : path)
+                fullPath.add(p);
+        }
+        return fullPath.toString();
+    }
+
+    ToXContentObject internalUserWithPassword(final String password) {
+        return (builder, params) -> builder.startObject()
+            .field("password", password)
+            .field("backend_roles", randomConfigArray(false))
+            .endObject();
+    }
+
+    @Test
+    public void canNotCreateUsersWithPassword() throws Exception {
+        withUser(ADMIN_USER_NAME, client -> {
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("")));
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("123")));
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("1234567")));
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("1Aa%")));
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("123456789")));
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("a123456789")));
+            badRequest(() -> client.putJson(internalUsers("tooshoort"), internalUserWithPassword("A123456789")));
+            badRequest(() -> client.putJson(internalUsers("$1aAAAAAAAAC"), internalUserWithPassword("$1aAAAAAAAAC")));
+            badRequest(() -> client.putJson(internalUsers("$1aAAAAAAAac"), internalUserWithPassword("$1aAAAAAAAAC")));
+            badRequest(
+                () -> client.putJson(
+                    internalUsers(URLEncoder.encode("$1aAAAAAAAac%!=\"/\\;:test&~@^", "UTF-8").replace("+", "%2B")),
+                    internalUserWithPassword("$1aAAAAAAAac%!=\\\"/\\\\;:test&~@^")
+                )
+            );
+            badRequest(
+                () -> client.putJson(
+                    internalUsers(URLEncoder.encode("$1aAAAAAAAac%!=\"/\\;: test&", "UTF-8")),
+                    internalUserWithPassword("$1aAAAAAAAac%!=\\\"/\\\\;: test&123")
+                )
+            );
+            badRequest(
+                () -> client.patch(
+                    internalUsers(),
+                    patch(
+                        addOp("testuser1", internalUserWithPassword("$aA123456789")),
```

Comment:
nit: Can you add minor comments regarding why this fails v/s same password works below?

Reply:
addressed

Reply:
@willyborankin Thanks :) However, I was thinking if we should add an in-line comment with the test suggesting reason for validation failure, thoughts?

Reply:
Add comments